plugins/http2_adapter/lib/src/connection_manager_imp.dart

```diff
@@ -111,6 +115,79 @@ class _ConnectionManager implements ConnectionManager {
     return _transportState;
   }
 
+  Future<SecureSocket> _createSocket(
+    Uri target,
+    RequestOptions options,
+    ClientSetting clientConfig,
+  ) async {
+    final socket = await SecureSocket.connect(
+      target.host,
+      target.port,
+      timeout: options.connectTimeout > 0
+          ? Duration(milliseconds: options.connectTimeout)
+          : null,
+      context: clientConfig.context,
+      onBadCertificate: clientConfig.onBadCertificate,
+      supportedProtocols: ['h2'],
+    );
+
+    return socket;
+  }
+
+  Future<SecureSocket> _createSocketWithProxy(
+    Uri target,
+    RequestOptions options,
+    ClientSetting clientConfig,
+    Uri proxy,
+  ) async {
+    late Socket proxySocket;
+
+    proxySocket = await Socket.connect(
+      proxy.host,
+      proxy.port,
+      timeout: options.connectTimeout > 0
+          ? Duration(milliseconds: options.connectTimeout)
+          : null,
+    );
+
+    //Send proxy headers
+    const crlf = '\r\n';
+    proxySocket.write('CONNECT ${target.host}:${target.port} HTTP/1.1');
+    proxySocket.write(crlf);
+    proxySocket.write('Host: ${target.host}:${target.port}');
+    proxySocket.write(crlf);
+    proxySocket.write(crlf);
+
+    final completer = Completer<void>();
+
+    final sub = proxySocket.listen(
+      (event) {
+        final response = ascii.decode(event);
+        final lines = response.split(crlf);
+        final statusLine = lines.first;
+        if (statusLine.startsWith('HTTP/1.1 200')) {
+          completer.complete();
+        } else {
+          completer.completeError(statusLine);
+        }
+      },
+      onError: completer.completeError,
+    );
+    await completer.future;
```

Comment:
Is the error case here intentionally not handled? 
I believe exceptions should be caught and rethrown as `DioError` like in the rest of the code base.

Reply:
Yes, all errors are caught above, but I added 'onError', is that better?